### PR TITLE
Docker: speed up docker build and remove unused files 

### PR DESCRIPTION
**Pull request by siddontang**
## What have you changed? (mandatory)

In our current Dockerfile, whenever we change the code, we need to rebuild all dependencies, it is very very slow.  

Refer to https://github.com/rust-lang/cargo/issues/2644, http://mgattozzi.com/caching-rust-docker-builds/, we know that there is a simple way to cache the dependencies, but for TiKV, we still need to do more.

Because TiKV uses multi-components, we must create a dummy component to let Docker cache the component's dependencies, so you can see that the Dockerfile is verbose and I also write a script to generate the Dockerfile like `./scripts/gen-dockerfile.sh`.

In my test with `docker build -t pingcap/tikv .` :

```
# Build dependencies
Finished release [optimized + debuginfo] target(s) in 21m 07s

# Build binaries
Finished release [optimized + debuginfo] target(s) in 12m 13s
```

After I change a line in TiKV, the build time is :

```
// no need to build dependencies now
// only build binaries
Finished release [optimized + debuginfo] target(s) in 11m 57s
```

We should know that even we can cache dependencies, we still need to do a full build for TiKV itself. So now the only thing for us is to optimize full build compiling time, /cc @brson 

Another thing is that now we don't use `docker` for releasing TiKV docker, the SRE team maintain `pingcap/rust` and `pingcap/tikv` by themselves, so we can remove this directory.

Here we also should know that every time when we add a new component, we need to rebuild the dockerfile through the script. 

## What are the type of the changes? (mandatory)

- Improvement (change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

Manually

## Does this PR affect documentation (docs) or release note? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

PTAL @brson @iamxy

### CODE REVIEW

**Comment by brson** [comment]
Good idea writing a script to write the dockerfile.

It's hard to evaluate because the whole thing is so obviously non-ideal and difficult to maintain, but I guess this is something we have to do.

It would be desirable if some CI script could detect that the dockerfile was out of date but I don't know how to do that.

**Comment by siddontang** [comment]
PTAL @iamxy

File: Dockerfile [hunk 1]
```diff
@@ -1,17 +1,67 @@
-# pingcap/rust: 2017-06-12
 FROM pingcap/rust as builder
 
-MAINTAINER Liu Yin <liuy@pingcap.com>
-
-COPY . /tikv
+RUN mkdir -p /tikv
```
Comment (iamxy) [inline Dockerfile]: The `WORKDIR` next line will create the directory, so it's not necessary.

File: scripts/gen-dockerfile.sh [hunk 2]
```diff
@@ -0,0 +1,82 @@
+# TiKV root 
+dir="."
+output="./Dockerfile"
+
+if [ "$#" -eq 1 ]; then
```
Comment (iamxy) [inline scripts/gen-dockerfile.sh]: Using `-ge` is better.

**Comment by siddontang** [comment]
PTAL @brson @iamxy

**Pull request merged**